rustls/src/client/tls13.rs

```diff
@@ -324,7 +324,9 @@ pub(super) fn fill_in_psk_binder(
     let real_binder = key_schedule.resumption_psk_binder_key_and_sign_verify_data(&handshake_hash);
 
     if let HandshakePayload::ClientHello(ch) = &mut hmp.payload {
-        ch.set_psk_binder(real_binder.as_ref());
+        if let Some(psk_offer) = &mut ch.extensions.preshared_key_offer {
+            psk_offer.binders[0] = PresharedKeyBinder::from(real_binder.as_ref().to_vec());
```

Comment:
Nit: document why `[0]` never panics?

Reply:
added a comment, and some debug asserts.